client_error.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Generic error thrown by the SDK.
+ * @extends {Error}
+ */
+class ClientError extends Error {
+  /**
+   * @override
+   * @param {StatusCode} statusCode error status code
+   * @param {string} args
+   */
+  constructor(statusCode, ...args) {
+    super(...args);
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, ClientError);
+    }
+    this.name = 'ClientError';
+    /**
+     * @type {StatusCode}
+     */
+    this.statusCode = statusCode;
```

Comment:
Reading a property of an object is normal in JavaScript, but I think have `getStatusCode` is better in this case because we:
1. Try to make the interface as similar to Java as we can.
2. More OO